# Project 4: Candidate Resume Recommendation System
# 01 Simple EDA

## 1. Import packages

In [3]:
import pandas as pd
from pymongo import MongoClient
from pprint import pprint
pd.set_option('display.max_colwidth', -1)

/Users/hellojenny/opt/anaconda3/envs/metis/lib/python3.7/site-packages/ipykernel_launcher.py:4: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  after removing the cwd from sys.path.


## 2. Read data
Data is downloaded [from Kaggle](https://www.kaggle.com/kandij/job-recommendation-datasets)

In [2]:
applicant_experience = pd.read_csv('/Users/hellojenny/Metis/Metis-Bootcamp/Projects/Project 4 - Unsupervised Learning & NLP/Data/applicant_experience.csv')
applicant_interests = pd.read_csv('/Users/hellojenny/Metis/Metis-Bootcamp/Projects/Project 4 - Unsupervised Learning & NLP/Data/applicant_interests.csv')
applicant_job_views = pd.read_csv('/Users/hellojenny/Metis/Metis-Bootcamp/Projects/Project 4 - Unsupervised Learning & NLP/Data/applicant_job_views.csv')
job_description = pd.read_csv('/Users/hellojenny/Metis/Metis-Bootcamp/Projects/Project 4 - Unsupervised Learning & NLP/Data/job_description.csv')
job_listing = pd.read_csv('/Users/hellojenny/Metis/Metis-Bootcamp/Projects/Project 4 - Unsupervised Learning & NLP/Data/job_listing.csv')

## 3. Understand each dataset
Since there are no descriptions on the datasets, we need to look at all the datasets to determine what they are and what we'd like to use for this project.

### Application Datasets

In [3]:
applicant_interests.head(5)

Applicant.ID Position.Of.Interest               Created.At  \
0  10003         security officer     2014-12-12 21:20:54 UTC   
1  10007         Server               2014-08-14 15:56:42 UTC   
2  10007         Bartender            2014-08-14 15:56:44 UTC   
3  10008         Host                 2014-08-14 15:56:42 UTC   
4  10008         Barista              2014-08-14 15:56:43 UTC   

                Updated.At  
0  2014-12-12 21:20:54 UTC  
1  2015-02-26 20:35:12 UTC  
2  2015-02-19 23:21:28 UTC  
3  2015-02-26 20:35:12 UTC  
4  2015-02-18 02:35:06 UTC

In [4]:
applicant_job_views.head(5)

Applicant.ID  Job.ID  \
0  10000         73666    
1  10000         96655    
2  10001         84141    
3  10002         77989    
4  10002         69568    

                                                                                    Title  \
0  Cashiers & Valets Needed! @ WallyPark                                                    
1  Macy's Seasonal Retail Fragrance Cashier  - Garden City, NY - Roosevelt Field @ Macy's   
2  Part Time Showroom Sales / Cashier @ Grizzly Industrial Inc.                             
3  Event Specialist Part Time @ Advantage Sales & Marketing                                 
4  Bonefish - Kitchen Staff @ Bonefish Grill                                                

                                                                        Position  \
0  Cashiers & Valets Needed!                                                       
1  Macy's Seasonal Retail Fragrance Cashier  - Garden City, NY - Roosevelt Field   
2  Part Time Showroom Sales / Cashier                                              
3  Event Specialist Part Time                                                      
4  Bonefish - Kitchen Staff                                                        

                       Company          City      State.Name State.Code  \
0  WallyPark                    Newark        New Jersey      NJ          
1  Macy's                       Garden City   New York        NY          
2  Grizzly Industrial Inc.      Bellingham    Washington      WA          
3  Advantage Sales & Marketing  Simpsonville  South Carolina  SC          
4  Bonefish Grill               Greenville    South Carolina  SC          

  Industry               View.Start                 View.End  View.Duration  \
0  NaN      2014-12-12 20:12:35 UTC  2014-12-12 20:31:24 UTC  1129.0          
1  NaN      2014-12-12 20:08:50 UTC  2014-12-12 20:10:15 UTC  84.0            
2  NaN      2014-12-12 20:12:32 UTC  2014-12-12 20:17:18 UTC  286.0           
3  NaN      2014-12-12 20:39:23 UTC  2014-12-12 20:42:13 UTC  170.0           
4  NaN      2014-12-12 20:43:25 UTC  2014-12-12 20:43:58 UTC  33.0            

                Created.At               Updated.At  
0  2014-12-12 20:12:35 UTC  2014-12-12 20:12:35 UTC  
1  2014-12-12 20:08:50 UTC  2014-12-12 20:08:50 UTC  
2  2014-12-12 20:12:32 UTC  2014-12-12 20:12:32 UTC  
3  2014-12-12 20:39:23 UTC  2014-12-12 20:39:23 UTC  
4  2014-12-12 20:43:25 UTC  2014-12-12 20:43:25 UTC

In [5]:
applicant_experience.head(5)

Applicant.ID                                               Position.Name  \
0  10001         Account Manager / Sales Administration / Quality Assurance   
1  10001         Electronics Technician / Item Master Controller              
2  10001         Machine Operator                                             
3  10003         maintenance technician                                       
4  10003         Electrical Helper                                            

               Employer.Name         City            State.Name State.Code  \
0  Barcode Resourcing         Bellingham   Washington            WA          
1  Ryzex Group                Bellingham   Washington            WA          
2  comptec inc                Custer       Washington            WA          
3  Winn residental            washington   District of Columbia  DC          
4  michael and son services   alexandria   Virginia              VA          

   Start.Date    End.Date  \
0  2012-10-15  NaN          
1  2001-12-01  2012-04-01   
2  1997-01-01  1999-01-01   
3  NaN         NaN          
4  NaN         NaN          

                                                      Job.Description  Salary  \
0  NaN                                                                NaN       
1  NaN                                                                NaN       
2  NaN                                                                NaN       
3  Necessary maintenance for "Make Ready" Plumbing, electrical, HvAc   10.0     
4  repair and services of electrical construction                     NaN       

  Can.Contact.Employer               Created.At               Updated.At  
0  NaN                  2014-12-12 20:10:02 UTC  2014-12-12 20:10:02 UTC  
1  NaN                  2014-12-12 20:10:02 UTC  2014-12-12 20:10:02 UTC  
2  NaN                  2014-12-12 20:10:02 UTC  2014-12-12 20:10:02 UTC  
3  False                2014-12-12 21:27:05 UTC  2014-12-12 21:27:05 UTC  
4  False                2014-12-12 21:27:05 UTC  2014-12-12 21:27:05 UTC

In [6]:
applicant_experience['Position.Name'].unique

<bound method Series.unique of 0       Account Manager / Sales Administration / Quality Assurance
1       Electronics Technician / Item Master Controller           
2       Machine Operator                                          
3       maintenance technician                                    
4       Electrical Helper                                         
               ...                                                
8648    Server                                                    
8649    server                                                    
8650    Server/Cocktail Server/Bartender                          
8651    Server/cocktail server/bartender                          
8652    Server                                                    
Name: Position.Name, Length: 8653, dtype: object>

### Job Listing Datasets

In [7]:
job_description.head(5)

Unnamed: 0  Job.ID  \
0  0           111      
1  1           113      
2  2           117      
3  3           121      
4  4           127      

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      text  
0  server tacolici palo alto part time tacolici first palo alto store open recent hire love taco love work restaur serv food drink custom profession manner act cashier need clean dine space train new staff                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
1  kitchen staff chef claud lane san francisco part time new french brasseri s.f. financi district seek chef seek energet dynam chef take charg grow company. ideal candid motiv self starter great work ethic readi challeng build team execut support. posit perfect fit talent chef readi take next step someon multi task high volum kitchen except organiz skills. posit requir experi french cuisin minimum year sou chef high volum full servic restaurant. must team player lead exampl work side side member team. comput manag skill must experi order inventori cost control. right candid previous demonstr understand leadership account infecti energet approach problem solv face challenges. equal opportun employer. qualifi applic consid employ without regard age race religion sex nation origin sexual orient disabl veteran status.  
2  bartend machka restaur corp. san francisco part time popular mediterranean wine bar restaur financi district. look experienc bartend dinner shift monday saturday . candid posit attitud good wine knowledg abl work team player take initi help whole floor serv wine beer only.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
3  server teriyaki hous brisban part time serv food drink custom profession manner act cashier need clean dine space train new staff japanes restaur brisban look servers.                                                                                                                                                                                                                                                                                                                                                                              

In [8]:
job_description.drop(columns=['Unnamed: 0'], inplace=True)
job_description.sort_values(by=['Job.ID']).head(5)

Job.ID  \
66565  3        
50456  28       
66566  30       
83985  33       
83992  35       

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        text  
66565  custom servic bayer healthcar cincinnati part time candid familiar microsoft offic comput general. duti includ limit greet client answer phone make travel arrang ceo scan file fax send emails. confidenti utmost importance. look someon detail orient profession image.                                                                                                                                                                                                                                                                                                                                                                                

In [9]:
job_listing.head(5)

Job.ID  Provider Status  \
0  111     1         open    
1  113     1         open    
2  117     1         open    
3  121     1         open    
4  127     1         open    

                                                     Slug  \
0  palo-alto-ca-tacolicious-server                          
1  san-francisco-ca-claude-lane-kitchen-staff-chef          
2  san-francisco-ca-machka-restaurants-corp-bartender       
3  brisbane-ca-teriyaki-house-server                        
4  los-angeles-ca-rosa-mexicano-sunset-kitchen-staff-chef   

                                         Title            Position  \
0  Server @ Tacolicious                         Server               
1  Kitchen Staff/Chef @ Claude Lane             Kitchen Staff/Chef   
2  Bartender @ Machka Restaurants Corp.         Bartender            
3  Server @ Teriyaki House                      Server               
4  Kitchen Staff/Chef @ Rosa Mexicano - Sunset  Kitchen Staff/Chef   

                    Company           City  State.Name State.Code  ...  \
0  Tacolicious               Palo Alto      California  CA         ...   
1  Claude Lane               San Francisco  California  CA         ...   
2  Machka Restaurants Corp.  San Francisco  California  CA         ...   
3  Teriyaki House            Brisbane       California  CA         ...   
4  Rosa Mexicano - Sunset    Los Angeles    California  CA         ...   

             Industry  \
0  Food and Beverages   
1  Food and Beverages   
2  Food and Beverages   
3  Food and Beverages   
4  Food and Beverages   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Job.Description  \
0  Tacolicious' first Palo Alto store just opened recently, and we are hiring! If you love tacos, you will love working at our restaurant! \r\n\r\n ● Serve food/drinks to customers in a professional manner \r\n ● Act as a cashier when needed \r\n ● Clean up the dining space \r\n ● Train the new staff \r\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 

In [10]:
job_listing.sort_values(by=['Job.ID']).head(5)

Job.ID  Provider Status  \
66565  3       1         open    
50456  28      1         open    
66566  30      1         open    
83985  33      1         open    
83992  35      1         open    

                                                    Slug  \
66565  cincinnati-oh-bayer-healthcare-customer-service     
50456  san-francisco-ca-pacific-catch-kitchen-staff-chef   
66566  olney-md-dave-s-american-bistro-bartender           
83985  oakland-ca-haven-server                             
83992  san-francisco-ca-skool-kitchen-staff                

                                     Title            Position  \
66565  Customer Service @ Bayer healthcare  Customer Service     
50456  Kitchen Staff/Chef @ Pacific Catch   Kitchen Staff/Chef   
66566  Bartender @ Dave's American Bistro   Bartender            
83985  Server @ Haven                       Server               
83992  Kitchen Staff @ Skool                Kitchen Staff        

                      Company           City  State.Name State.Code  ...  \
66565  Bayer healthcare        Cincinnati     Ohio        OH         ...   
50456  Pacific Catch           San Francisco  California  CA         ...   
66566  Dave's American Bistro  Olney          Maryland    MD         ...   
83985  Haven                   Oakland        California  CA         ...   
83992  Skool                   San Francisco  California  CA         ...   

                    Industry  \
66565  Office Administration   
50456  Food and Beverages      
66566  Food and Beverages      
83985  Food and Beverages      
83992  Food and Beverages      

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          

### Joining datasets to create the final `job_posting` dataframe
It seems that `job_listing` dataset and `job_description` dataset can be joined on `Job.ID`. 

In [11]:
job_posting = job_listing.join(job_description, lsuffix='_listing', rsuffix='_description')
job_posting.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84090 entries, 0 to 84089
Data columns (total 25 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Job.ID_listing      84090 non-null  int64  
 1   Provider            84090 non-null  int64  
 2   Status              84090 non-null  object 
 3   Slug                84090 non-null  object 
 4   Title               84090 non-null  object 
 5   Position            84090 non-null  object 
 6   Company             81819 non-null  object 
 7   City                83955 non-null  object 
 8   State.Name          83919 non-null  object 
 9   State.Code          83919 non-null  object 
 10  Address             36 non-null     object 
 11  Latitude            84090 non-null  float64
 12  Longitude           84090 non-null  float64
 13  Industry            267 non-null    object 
 14  Job.Description     84034 non-null  object 
 15  Requirements        0 non-null      float64
 16  Sala

Here's the joined `job_posting` dataset.

In [12]:
job_posting.head(5)

Job.ID_listing  Provider Status  \
0  111             1         open    
1  113             1         open    
2  117             1         open    
3  121             1         open    
4  127             1         open    

                                                     Slug  \
0  palo-alto-ca-tacolicious-server                          
1  san-francisco-ca-claude-lane-kitchen-staff-chef          
2  san-francisco-ca-machka-restaurants-corp-bartender       
3  brisbane-ca-teriyaki-house-server                        
4  los-angeles-ca-rosa-mexicano-sunset-kitchen-staff-chef   

                                         Title            Position  \
0  Server @ Tacolicious                         Server               
1  Kitchen Staff/Chef @ Claude Lane             Kitchen Staff/Chef   
2  Bartender @ Machka Restaurants Corp.         Bartender            
3  Server @ Teriyaki House                      Server               
4  Kitchen Staff/Chef @ Rosa Mexicano - Sunset  Kitchen Staff/Chef   

                    Company           City  State.Name State.Code  ...  \
0  Tacolicious               Palo Alto      California  CA         ...   
1  Claude Lane               San Francisco  California  CA         ...   
2  Machka Restaurants Corp.  San Francisco  California  CA         ...   
3  Teriyaki House            Brisbane       California  CA         ...   
4  Rosa Mexicano - Sunset    Los Angeles    California  CA         ...   

  Requirements  Salary  Listing.Start Listing.End Employment.Type  \
0 NaN           8.00    NaN            NaN         Part-Time        
1 NaN           0.00    NaN            NaN         Part-Time        
2 NaN           11.00   NaN            NaN         Part-Time        
3 NaN           10.55   NaN            NaN         Part-Time        
4 NaN           10.55   NaN            NaN         Part-Time        

   Education.Required               Created.At               Updated.At  \
0  NaN                 2013-03-12 02:08:28 UTC  2014-08-16 15:35:36 UTC   
1  NaN                 2013-04-12 08:36:36 UTC  2014-08-16 15:35:36 UTC   
2  NaN                 2013-07-16 09:34:10 UTC  2014-08-16 15:35:37 UTC   
3  NaN                 2013-09-04 15:40:30 UTC  2014-08-16 15:35:38 UTC   
4  NaN                 2013-07-17 15:26:18 UTC  2014-08-16 15:35:40 UTC   

  Job.ID_description  \
0  111                 
1  113                 
2  117                 
3  121                 
4  127                 

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      text  
0  server tacolici palo alto part time tacolici first palo alto store open recent hire love taco love work restaur serv food drink custom profession manner act cashier need clean dine space train new staff                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                

### Renaming columns for dataframes
Finally, we'll rename the columns for the datasets that we'll use later. 

In [13]:
jobs_final = job_posting.loc[:, ['Job.ID_listing', 'Slug', 'Position', 'Company', 'Industry', 'Job.Description',
                     'Employment.Type', 'Education.Required', 'text']]

jobs_final.rename(columns = {'Job.ID_listing': 'job_id',
                             'Slug': 'slug',
                             'Position': 'job_title',
                             'Company': 'company',
                             'Industry': 'industry',
                             'Job.Description': 'job_description',
                             'Employment.Type': 'employment_type',
                             'Education.Required': 'education', 
                             'text': 'org_cleaned_text'}, inplace=True)

In [14]:
resumes_final = applicant_experience.loc[:, ['Applicant.ID', 'Position.Name', 'Employer.Name', 'Job.Description', 'Salary']]

resumes_final.rename(columns = {'Applicant.ID': 'applicant_id',
                                'Position.Name': 'job_title', 
                                'Employer.Name': 'employer', 
                                'Job.Description': 'job_description',
                                'Salary': 'salary'}, inplace=True)

## 4. Pickle datasets that will be used later

In [15]:
jobs_final.to_pickle('./jobs.pkl')
resumes_final.to_pickle('./resumes.pkl')

## 5. Conduct simple EDA on the final datasets

### Job Descriptions

In [7]:
jobs_final = pd.read_pickle('./jobs.pkl')
jobs_final.iloc[:, 0:7].head(11)

job_id  \
0   111      
1   113      
2   117      
3   121      
4   127      
5   129      
6   131      
7   133      
8   134273   
9   134274   
10  134275   

                                                                                                                  slug  \
0   palo-alto-ca-tacolicious-server                                                                                      
1   san-francisco-ca-claude-lane-kitchen-staff-chef                                                                      
2   san-francisco-ca-machka-restaurants-corp-bartender                                                                   
3   brisbane-ca-teriyaki-house-server                                                                                    
4   los-angeles-ca-rosa-mexicano-sunset-kitchen-staff-chef                                                               
5   los-altos-ca-mind-of-beauty-day-spa-receptionist                                                                     
6   los-angeles-ca-roy-s-woodland-hills-server                                                                           
7   berkeley-ca-koja-kitchen-driver                                                                                      
8   mendham-nj-king-s-food-markets-assistant-store-manager                                                               
9   onalaska-wi-aldi-store-associate-retail-sales-customer-service                                                       
10  skokie-il-macy-s-macy-s-seasonal-retail-commission-sales-women-s-shoes-part-time-skokie-il-old-orchard-mall-134275   

                                                                                           job_title  \
0   Server                                                                                             
1   Kitchen Staff/Chef                                                                                 
2   Bartender                                                                                          
3   Server                                                                                             
4   Kitchen Staff/Chef                                                                                 
5   Receptionist                                                                                       
6   Server                                                                                             
7   Driver                                                                                             
8   Assistant Store Manager                                                                            
9   Store Associate - Retail Sales (Customer Service)                                                  
10  Macy's Seasonal Retail Commission Sales Women's Shoes, Part Time - Skokie, IL - Old Orchard Mall   

                     company            industry  \
0   Tacolicious               Food and Beverages   
1   Claude Lane               Food and Beverages   
2   Machka Restaurants Corp.  Food and Beverages   
3   Teriyaki House            Food and Beverages   
4   Rosa Mexicano - Sunset    Food and Beverages   
5   Mind of Beauty Day Spa    Retail               
6   Roy's Woodland Hills      Food and Beverages   
7   KoJa Kitchen              Food and Beverages   
8   King's Food Markets       NaN                  
9   ALDI                      NaN                  
10  Macy's                    NaN                  

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                

In [6]:
jobs_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84090 entries, 0 to 84089
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   job_id            84090 non-null  int64 
 1   slug              84090 non-null  object
 2   job_title         84090 non-null  object
 3   company           81819 non-null  object
 4   industry          267 non-null    object
 5   job_description   84034 non-null  object
 6   employment_type   84080 non-null  object
 7   education         83823 non-null  object
 8   org_cleaned_text  81766 non-null  object
dtypes: int64(1), object(8)
memory usage: 5.8+ MB


In [18]:
len(jobs_final.job_title.unique())

36245

In [19]:
len(jobs_final.industry.unique())

7

In [20]:
jobs_final.industry.value_counts()

Food and Beverages       152
Other                    44 
Transportation           39 
Retail                   21 
Office Administration    10 
Care Giving              1  
Name: industry, dtype: int64

In [22]:
jobs_final.employment_type.value_counts()

Part-Time              33228
Seasonal/Temp          27609
Full-Time/Part-Time    17626
Per Diem               4644 
Intern                 921  
Full-Time              37   
Contract               14   
Temporary/seasonal     1    
Name: employment_type, dtype: int64

The `jobs_final` datasets has 84,090 job listings, and there are lots of values missing for the below columns that we will not 

In [24]:
print([i for i in jobs_final.job_title.unique()])

['Server', 'Kitchen Staff/Chef', 'Bartender', 'Receptionist', 'Driver', 'Assistant Store Manager', 'Store Associate - Retail Sales (Customer Service)', "Macy's Seasonal Retail Commission Sales Women's Shoes, Part Time - Skokie, IL - Old Orchard Mall", 'Part Time Clerks Needed', 'Flex (Part Time) Merchandiser - Mullins, SC', 'Child Psychotherapist', 'Hiring Kitchen Positions - Experienced Line Cooks', 'HR Administrator', 'Retail Gift Registry Advisor, Full Time/Part Time: Dayton, OH - Macy’s Dayton Mall', 'Weekend Breakfast Attendant', 'EMT - Growing Medical Transportation Company', 'Part-Time Financial Operations Specialist (Undergraduate Student Program)', 'Macy’s Seasonal Retail Stock Merchandising Part Time, Morning 5am-2pm - San Jose, CA - Oakridge', 'Seasonal Retail Selling Floor Recovery, As Early As 11AM - San Mateo, CA -  Hillsdale Shopping Center', 'Retail Administrative Support Team, Part Time: Westland, MI - Macy’s Westland Center', 'Teaching Opportunties', 'Sales Profession

### Resumes

Each row is one experience for an applicant. As an example, `applicant_id` 10001 has 3 experiences, hence the first 3 rows are all his experiences. 

In [25]:
resumes_final.head(5)

applicant_id                                                   job_title  \
0  10001         Account Manager / Sales Administration / Quality Assurance   
1  10001         Electronics Technician / Item Master Controller              
2  10001         Machine Operator                                             
3  10003         maintenance technician                                       
4  10003         Electrical Helper                                            

                    employer  \
0  Barcode Resourcing          
1  Ryzex Group                 
2  comptec inc                 
3  Winn residental             
4  michael and son services    

                                                      job_description  salary  
0  NaN                                                                NaN      
1  NaN                                                                NaN      
2  NaN                                                                NaN      
3  Necessary maintenance for "Make Ready" Plumbing, electrical, HvAc   10.0    
4  repair and services of electrical construction                     NaN

In [26]:
resumes_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8653 entries, 0 to 8652
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   applicant_id     8653 non-null   int64  
 1   job_title        7703 non-null   object 
 2   employer         8620 non-null   object 
 3   job_description  5692 non-null   object 
 4   salary           2798 non-null   float64
dtypes: float64(1), int64(1), object(3)
memory usage: 338.1+ KB


In [27]:
len(resumes_final.applicant_id.unique())

3790

In [28]:
resumes_final['job_description'][20:25]

20    Greet customers in friendly manner, handled cash in cash register, rotated stock, count inventory, answer phones.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 